### PR TITLE
Replace lastEventId GET parameter in Request-URL if it's already definded

### PR DESCRIPTION
**Pull request by marioklump**
I'm using [mercure](https://mercure.rocks/) as SEE server.

To get all messages the user "missed" while the app was closed, I append the last event ID to the initialization of `EventSourcePolyfill`. In my case, the value of `lastEventId` is stored in the local storage of the app.

```typescript
const url = new URL(eventStreamUrl);
url.searchParams.append('topic', 'chat-messages');

// Append to ID of the last fetched event to the request to get all events missed while the app was closed.
url.searchParams.append('Last-Event-ID', lastEventId);

this.eventSource = new EventSourcePolyfill(url, {
  lastEventIdQueryParameterName: 'Last-Event-ID',
  headers: {
    Authorization: `Bearer ...`
  }
});
```

When EventSource is reconnecting to the SSE server, the Last-Event-ID known to EventSourcePolyfill is appended to the request URL. This causes the parameter `Last-Event-ID` to be contained in the request URL twice:

```
https://SSE-SERVER/.well-known/mercure?topic=chat-messages&Last-Event-ID=urn%3Auuid%3A...AAA&Last-Event-ID=urn%3Auuid%3A...BBB
```

This causes failed reconnects to the SSE server because of the duplicate GET parameters.
That's why I changed the code to use `URLSearchParams.set(...)`  to set/add the parameter `Last-Event-ID`. If the parameter `Last-Event-ID` is already there (for example because of my `EventSourcePolyfill` initialization), it will be replaced by the latest `Last-Event-ID`.

### CODE REVIEW

**Comment by marioklump** [comment]
@Yaffle should we merge this pull request?

**Comment by Yaffle** [comment]
@marioklump , the URL constructor is not supported by old browsers....

File: src/eventsource.js [hunk 1]
```diff
@@ -983,7 +983,7 @@
         }
       }
       try {
-        abortController = transport.open(xhr, onStart, onProgress, onFinish, requestURL, withCredentials, requestHeaders);
+        abortController = transport.open(xhr, onStart, onProgress, onFinish, requestURL.toString(), withCredentials, requestHeaders);
```
Comment (Yaffle) [inline src/eventsource.js:986]:
```suggestion
        abortController = transport.open(xhr, onStart, onProgress, onFinish, requestURL, withCredentials, requestHeaders);
```

**Review by Yaffle — CHANGES_REQUESTED** [review]

File: src/eventsource.js [hunk 2]
```diff
@@ -965,10 +965,10 @@
 
       // https://bugzilla.mozilla.org/show_bug.cgi?id=428916
       // Request header field Last-Event-ID is not allowed by Access-Control-Allow-Headers.
-      var requestURL = url;
+      var requestURL = new URL(url);
       if (url.slice(0, 5) !== "data:" && url.slice(0, 5) !== "blob:") {
         if (lastEventId !== "") {
-          requestURL += (url.indexOf("?") === -1 ? "?" : "&") + lastEventIdQueryParameterName +"=" + encodeURIComponent(lastEventId);
+          requestURL.searchParams.set(lastEventIdQueryParameterName, lastEventId);
```
Comment (Yaffle) [inline src/eventsource.js:971]:
```suggestion
          var i = url.indexOf("?");
          requestURL = i === -1 ? url : url.slice(0, i + 1) + url.slice(i + 1).replace(/(?:^|&)([^=&]*)(?:=[^&]*)?/g, function (p, paramName) {
            return paramName === lastEventIdQueryParameterName ? '' : p;
          });
          requestURL += (url.indexOf("?") === -1 ? "?" : "&") + lastEventIdQueryParameterName +"=" + encodeURIComponent(lastEventId);
```

**Review by Yaffle — CHANGES_REQUESTED** [review]

**Comment by marioklump** [comment]
@Yaffle Thank you for your suggestions. I've tested it on my local machine and it looks fine.

File: src/eventsource.js [hunk 3]
```diff
@@ -968,6 +968,12 @@
       var requestURL = url;
       if (url.slice(0, 5) !== "data:" && url.slice(0, 5) !== "blob:") {
         if (lastEventId !== "") {
+          // Remove the lastEventId parameter if it's already part of the request URL.
```
Comment (Yaffle) [inline src/eventsource.js:971]: good comments!

**Review by Yaffle — APPROVED** [review]

**Pull request merged**